convert2rhel/repo.py

```diff
@@ -47,3 +54,57 @@ def get_rhel_repoids():
     loggerinst.info("RHEL repository IDs to enable: %s" % ", ".join(repos_needed))
 
     return repos_needed
+
+
+def download_repofile(repofile_url):
+    """Download a repository file from a specific URL.
+
+    :raises exceptions.CriticalError: When the repository file URL is inaccessible.
+    :returns str: Contents of the repofile if successfully downloaded.
+    """
+    try:
+        with closing(urllib.request.urlopen(repofile_url, timeout=15)) as response:
+            contents = response.read()
+
+            if not contents:
+                loggerinst.critical_no_exit("")
```

Comment:
I suppose you wanted to pass some text here.

Reply:
oops, that's right. thanks for spotting that

Reply:
Added the same message as the exception description